packages/flutter/lib/src/rendering/view.dart

```diff
@@ -303,20 +308,42 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
       case TargetPlatform.windows:
         break;
     }
-    // If there are no overlay styles in the UI don't bother updating.
-    if (upperOverlayStyle != null || lowerOverlayStyle != null) {
+    // If there are no overlay style in the UI don't bother updating.
+    if (upperOverlayStyle == null && lowerOverlayStyle == null) {
+      return;
+    }
+
+    // If both are not null, the upper provides the status bar properties and the lower provides
+    // the system navigation bar properties.
```

Comment:
Readability: Could you provide a brief explanation for why we do this?

Reply:
> Readability: Could you provide a brief explanation for why we do this?

Sorry, do you mean by updating this comment in `view.dart` or by commenting here on Github?

Reply:
I think here in `view.dart`, although the comments explaining `top` and `bottom` above are helpful. Maybe just providing an example of why there would be more than one, an upper and a lower?